target_decisioning_engine/geo_provider.py

```diff
@@ -0,0 +1,188 @@
+# Copyright 2021 Adobe. All rights reserved.
+# This file is licensed to you under the Apache License, Version 2.0 (the "License")
+# you may not use this file except in compliance with the License. You may obtain a copy
+# of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software distributed under
+# the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+# OF ANY KIND, either express or implied. See the License for the specific language
+# governing permissions and limitations under the License.
+"""GeoProvider class and related functions"""
+# pylint: disable=broad-except
+# pylint: disable=too-few-public-methods
+
+try:
+    from functools import reduce
+except ImportError:
+    pass
+import json
+from copy import deepcopy
+import urllib3
+from delivery_api_client import Geo
+from target_tools.logger import get_logger
+from target_tools.utils import noop
+from target_decisioning_engine.constants import HTTP_GET
+from target_decisioning_engine.constants import OK
+from target_decisioning_engine.events import GEO_LOCATION_UPDATED
+from target_decisioning_engine.utils import get_geo_lookup_path
+from target_decisioning_engine.constants import HTTP_HEADER_FORWARDED_FOR
+from target_decisioning_engine.constants import HTTP_HEADER_GEO_LATITUDE
+from target_decisioning_engine.constants import HTTP_HEADER_GEO_LONGITUDE
+from target_decisioning_engine.constants import HTTP_HEADER_GEO_COUNTRY
+from target_decisioning_engine.constants import HTTP_HEADER_GEO_REGION
+from target_decisioning_engine.constants import HTTP_HEADER_GEO_CITY
+from target_python_sdk.utils import parse_float
+
+
+def no_parse(value):
+    """Returns input as-is"""
+    return value
+
+
+GEO_MAPPINGS = [
+    {
+        "header_name": HTTP_HEADER_FORWARDED_FOR,
+        "parse_value": no_parse,
```

Comment:
as we're only ever parsing float, maybe it would be better to replace this with a bool
```suggestion
        "parse_float": False,
```